src/PlanOnlyStrategy.groovy

```diff
@@ -0,0 +1,37 @@
+import static TerraformEnvironmentStage.ALL
+import static TerraformEnvironmentStage.PLAN
+
+class PlanOnlyStrategy {
+
+    private TerraformInitCommand initCommand
+    private TerraformPlanCommand planCommand
+    private Jenkinsfile jenkinsfile
+
+    public Closure createPipelineClosure(String environment, StageDecorations decorations, List params) {
+        initCommand = TerraformInitCommand.instanceFor(environment)
+        planCommand = TerraformPlanCommand.instanceFor(environment)
+
+        jenkinsfile = Jenkinsfile.instance
+
+        return { ->
+            node(jenkinsfile.getNodeName()) {
+                deleteDir()
+                checkout(scm)
+                properties([parameters(params)])
+
+                decorations.apply(ALL) {
+                    stage("${PLAN}-${environment}") {
+                        decorations.apply(PLAN) {
+                            sh initCommand.toString()
+                            def status = sh(returnStatus: true, script: "set -o pipefail;" + planCommand.toString())
+                            if ( status != 0 ) {
+                                echo "ERROR: Expected NO CHANGES to terraform resources."
+                                currentBuild.result = 'FAILURE'
+                            }
+                        }
+                    }
```

Comment:
Can this be implemented as a decoration around Plan?

We can also skip the apply in the same way that the ConditionalApplyPlugin does.  These approaches don't require us to alter the internal  structure of TerraformEnvironmentStage.

Reply:
I'll have to play around to see if this can be a decoration around Plan. The problem with that is I need to capture the exit code from the plan command.